src/lib/utils/functions/RegularExpressions.ts

```diff
@@ -0,0 +1,18 @@
+// doi regex here - https://www.crossref.org/blog/dois-and-matching-regular-expressions/
+// note - had to add an escape character for the second and third forward slash in the regex above
+export const DOI_REGEX = /^10.\d{4,9}\/[-._;()\/:a-z0-9]+$/i
+
+/**
+ * Checks for a Synapse ID, with or without a version number.
+ * Captures the synId and version number into capture groups.
+ *
+ * Usage examples:
+ * > const result = 'syn123.9'.match(SYNAPSE_ENTITY_ID_REGEX)
+ * > result[0]
+ * 'syn123.9'
+ * > result[1]
+ * 'syn123'
+ * > result[2]
+ * '9'
+ */
+export const SYNAPSE_ENTITY_ID_REGEX = /^(syn\d+)(?:\.(\d+))?$/
```

Comment:
I changed this regex to also utilize capture groups to find the entity ID and version number. Functionality in existing places is unchanged. Usage examples in the comment